Classes/Domain/Repository/CommentRepository.php

```diff
@@ -246,12 +246,23 @@ protected function getFindByEmailAndPostIdConstraints(QueryInterface $query, $em
      *
      * @return object
      */
-    protected function getValidConstraints(QueryInterface $query)
+    protected function getValidConstraints(QueryInterface $query, $excludeHiddenPosts = false)
     {
         $constraints = $query->logicalAnd(
             $query->equals('spam', 0),
             $query->equals('approved', 1)
         );
+        // By default, comments of hidden posts will be included in the result, because there is no relation / join from comments to posts
```

Comment:
The result will include comments of hidden posts, but the `$comment->post` relation will be `null` when the post is hidden.